src/middlewares/createRouterMiddleware.ts

```diff
@@ -78,6 +76,11 @@ export const createRouterMiddleware = <
       },
 
       subscribe() {
+        instantSearchInstance._initialUiState = {
+          ...initialUiState,
```

Comment:
this can continue to be read from the InstantSearch instance right?

Reply:
But if subscribe() gets called multiple times, it could accidentally accumulate states.

For example, 

1. /search?hitsPerPage=3&query=abc

2. /search?query=def

but at the second visit, it will still get `hitsPerPage=3`, which is wrong.